pkg/operator/ceph/cluster/watcher.go

```diff
@@ -193,3 +197,83 @@ func (c *clientCluster) getCephCluster() *cephv1.CephCluster {
 
 	return &clusterList.Items[0]
 }
+
+// each OSD nearfull status is checked
+// OSD capacity is increased if we get nearfull warning
+func increaseOSDsCapacity(cephStatusChecker *cephStatusChecker) {
+	osdUsage, err := cephclient.GetOSDUsage(cephStatusChecker.context, cephStatusChecker.clusterInfo)
+	if err != nil {
+		logger.Debugf("failed to get osd usage")
+	}
+	// by default nearfull_ratio is set to 85%
+	osdNearfullRatio := 85
+
+	var osdsNearfullList []int
+	for _, osdStatus := range osdUsage.OSDNodes {
+		osdId := osdStatus.ID
+		storageUtilization, err := osdStatus.Utilization.Int64()
+		if err == nil {
+			if storageUtilization > int64(osdNearfullRatio) {
+				osdsNearfullList = append(osdsNearfullList, osdId)
+			}
+		}
+
+	}
+	increaseDeviceSetCapacity(cephStatusChecker, osdsNearfullList)
+}
+
+// list the storageClassDeviceSets that are linked to OSDs, and increase there storageSize
+func increaseDeviceSetCapacity(cephStatusChecker *cephStatusChecker, osdsNearfullList []int) {
+	var storageClassDeviceSetsList = util.NewSet()
```

Comment:
we don't need to use `var` if you are using the var inside the function. same below